app/main/services/query_builder.py

```diff
@@ -73,34 +166,26 @@ def match_all_clause():
 
 
 def filter_clause(query_args):
-    return {
-        "bool": {
-            "must": build_term_filters(query_args)
-        }
+    and_filters = []
+    or_filters = []
+
+    query_filters = [QueryFilter(field, values)
+                     for field, values in query_args.lists()
+                     if field.startswith("filter")]
+    filters = {
+        "bool": {}
     }
 
+    for each_filter in query_filters:
+        if each_filter.is_and_filter():
+            and_filters += each_filter.terms()
+        if each_filter.is_or_filter():
+            or_filters += each_filter.terms()
 
-def build_term_filters(query_args):
-    must = []
-    if "serviceTypes" in query_args:
-        for service_type in extract_service_types(query_args):
-            must.append({
-                "term": {
-                    "serviceTypesExact":
-                        strip_and_lowercase(service_type)
-                }
-            })
-    if "lot" in query_args:
-        must.append({
-            "term": {
-                "lot": query_args["lot"]
-            }
-        })
-    return must
+    if and_filters:
+        filters["bool"]["must"] = and_filters
 
+    if or_filters:
+        filters["bool"]["should"] = or_filters
```

Comment:
"should" is not strictly a filter, it just adds relevance to the results. I believe what we need here is a list of "must" clauses which include the AND filters as is and for each OR filter includes a `{"bool": {"should": [ ... terms ...], "minimum_should_match": 1}`.